model-optimizer/extensions/back/MarkNodesWithShapeValues.py

```diff
@@ -0,0 +1,88 @@
+"""
+ Copyright (C) 2018-2021 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import logging as log
+
+import numpy as np
+
+from extensions.middle.MarkSubgraphsWithCorrectLayout import MarkSubGraphsWithCorrectLayout
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph
+
+
+class MarkNodesWithShapeValues(BackReplacementPattern):
+    """
+    This replacer marks op nodes in ShapeOf subgraphs with 'returns_shape_value' bool attribute and
+    data nodes of float32 constants with 'correct_data_type' attribute.
+    So that float Consts and Cast_to float will be kept in FP32 even if cmd_param --data_type=FP16 was specified.
```

Comment:
"Cast_to" ? Please, change.

"cmd_param"? Please, change

"was" -> "is"

Reply:
fixed